### PR TITLE
THOR-948 non required fields

### PR DESCRIPTION
**Pull request by arciisine**
Making sortField optional, and handling the fallout from those changes in ThumbnailGrid, DataTable, and NewsFeed components.

### CODE REVIEW

File: karma-dev.conf.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module.exports = function (config) {
```
Comment (ThomasSchellenbergNextCentury) [inline karma-dev.conf.js:1]: What's up with this new karma conf file?

Comment (arciisine) [reply karma-dev.conf.js:1]: This is to allow for using `ng test` in the interactive mode to facilitate iterative testing/development.  This is a copy of a default karma.config provided by `ng new`.

**Comment by ThomasSchellenbergNextCentury** [comment]
As we discussed yesterday, I created a ticket for adding a default sort-by-ID in the Neon Server if no sort is specified:  https://nextcentury.atlassian.net/browse/THOR-1159

**Comment by ThomasSchellenbergNextCentury** [comment]
I've created a PR to fix the issue we discussed yesterday with having to add empty filters to the queries:  https://github.com/NextCenturyCorporation/neon-dash-internal/pull/205

File: karma-dev.conf.js (hunk 1 above)
Comment (ThomasSchellenbergNextCentury) [reply karma-dev.conf.js:1]: Oh, that sounds interesting.  You'll have to give me a demo sometime!

**Review by rartiss55 — DISMISSED** [review]

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]

**Review by rartiss55 — APPROVED** [review]

**Pull request merged**